src/einsteinpy/coordinates/utils.py

```diff
@@ -6,193 +6,193 @@
 _c = constant.c.value
 
 
-def cartesian_to_spherical_fast(
-    t, x, y, z, v_x=None, v_y=None, v_z=None, velocities_provided=False
-):
-    if velocities_provided:
-        return cartesian_to_spherical(t, x, y, z, v_x, v_y, v_z)
-    return cartesian_to_spherical_novel(t, x, y, z)
+@jit
+def calculate_coordinates(r, th, p, alpha):
+    xa = np.sqrt(r**2 + alpha**2)
+    sin_norm = xa * np.sin(th)
+    e0 = sin_norm * np.cos(p)
+    e1 = sin_norm * np.sin(p)
+    e2 = r * np.cos(th)
+    return xa, e0, e1, e2
+
+
+def cartesian_to_spherical_fast(e0, e1, e2, e3, u0=None, u1=None, u2=None):
+    vel = [u0, u1, u2]
+    if all(u is not None for u in vel) and any(u != 0 for u in vel):
+        return cartesian_to_spherical(e0, e1, e2, e3, u0, u1, u2)
+    return cartesian_to_spherical_novel(e0, e1, e2, e3)
 
 
 @jit
-def cartesian_to_spherical(t, x, y, z, v_x, v_y, v_z):
+def cartesian_to_spherical(e0, e1, e2, e3, u0, u1, u2):
     """
     Utility function (jitted) to convert cartesian to spherical.
     This function should eventually result in Coordinate Transformation Graph!
 
     """
-    hxy = np.hypot(x, y)
-    r = np.hypot(hxy, z)
-    theta = np.arctan2(hxy, z)
-    phi = np.arctan2(y, x)
-    n1 = x**2 + y**2
-    n2 = n1 + z**2
-    v_r = (x * v_x + y * v_y + z * v_z) / np.sqrt(n2)
-    v_th = (z * (x * v_x + y * v_y) - n1 * v_z) / (n2 * np.sqrt(n1))
-    v_p = -1 * (v_x * y - x * v_y) / n1
+    e4 = np.hypot(e1, e2)
+    e5 = np.hypot(e4, e3)
+    e6 = np.arctan2(e4, e3)
+    e7 = np.arctan2(e2, e1)
+    n1 = e1**2 + e2**2
+    n2 = n1 + e3**2
+    u3 = (e1 * u0 + e2 * u1 + e3 * u2) / np.sqrt(n2)
+    u4 = (e3 * (e1 * u0 + e2 * u1) - n1 * u2) / (n2 * np.sqrt(n1))
+    u5 = -1 * (u0 * e2 - e1 * u1) / n1
 
-    return t, r, theta, phi, v_r, v_th, v_p
+    return e0, e5, e6, e7, u3, u4, u5
 
 
 @jit
-def cartesian_to_spherical_novel(t, x, y, z):
+def cartesian_to_spherical_novel(e0, e1, e2, e3):
     """
     Utility function (jitted) to convert cartesian to spherical.
     This function should eventually result in Coordinate Transformation Graph!
 
     """
-    hxy = np.hypot(x, y)
-    r = np.hypot(hxy, z)
-    theta = np.arctan2(hxy, z)
-    phi = np.arctan2(y, x)
+    e4 = np.hypot(e1, e2)
+    e5 = np.hypot(e4, e3)
+    e6 = np.arctan2(e4, e3)
+    e7 = np.arctan2(e2, e1)
 
-    return t, r, theta, phi
+    return e0, e5, e6, e7
 
 
-def cartesian_to_bl_fast(
-    t, x, y, z, alpha, v_x=None, v_y=None, v_z=None, velocities_provided=False
-):
-    if velocities_provided:
-        return cartesian_to_bl(t, x, y, z, alpha, v_x, v_y, v_z)
-    return cartesian_to_bl_novel(t, x, y, z, alpha)
+def cartesian_to_bl_fast(e0, e1, e2, e3, e4, u0=None, u1=None, u2=None):
+    vel = [u0, u1, u2]
+    if all(u is not None for u in vel) and any(u != 0 for u in vel):
+        return cartesian_to_bl(e0, e1, e2, e3, e4, u0, u1, u2)
```

Comment:
`e4` is actually `u0` (`v_t`). Refer to the comment in `src/einsteinpy/coordinates/conversion.py`.